### PR TITLE
Updating PIP-1

### PR DESCRIPTION
**Pull request by jhajduk-microsoft**
# Overview/Summary

Updating PIP-1 with ARG query

- [X] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [X] Ensured PR tests are passing
- [X] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @jhajduk-microsoft! Can you please include a check in the query for Standard SKU IPs that are not zone-redundant, and include a screenshot of your query test results in the PR?

File: docs/content/services/networking/public-ip/code/pip-1/pip-1.kql
Comment (ehaslett) [inline docs/content/services/networking/public-ip/code/pip-1/pip-1.kql:1]: This is good to check if a PIP is not a standard SKU and report this, but it should also check PIPs that are Standard SKU but not zone-redundant and present those as well.

**Review by ehaslett — CHANGES_REQUESTED** [review]

**Comment by tksh164** [comment]
@jhajduk-microsoft @ejhenry - The ARG query looks the same as [pip-1.kql.fix](https://github.com/Azure/Azure-Proactive-Resiliency-Library/blob/main/docs/content/services/networking/public-ip/code/pip-1/pip-1.kql.fix) file. Also, PIP-1 recommendation revised on #301 

Standard Public IPs have possibility that non-zonal and zonal even if Standard SKU.
- [Test case for PIP-1](https://github.com/tksh164/aprl-testing/tree/main/services/networking/public-ip/test/pip-1)

On other hand, we should add a new recommendation for "Upgrade Basic SKU to Standard SKU".

- [Upgrade a public IP address using the Azure portal](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-upgrade-portal)

**Comment by jhajduk-microsoft** [comment]
@tksh164 - I thought it was odd that the query already existed. I am not part of the team that maintains this so not sure what the fix files are for so did not know the ask was revised. I will add the zonal check

**Comment by jhajduk-microsoft** [comment]
image

**Review by ejhenry — COMMENTED** [review]
The query as currently written does not return Standard sku PIPs that are not zone-redundant.  We need to catch these as well to identify them for customers as not being aligned with the recommendation.

**Comment by tksh164** [comment]
@jhajduk-microsoft - FYI regarding `.fix` file.

- [Azure Resource Graph (ARG) Queries - Contribution guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing/#azure-resource-graph-arg-queries)
> 5. If a ARG query folder has a file with a file type suffixed with .fix, this means that the current query does not work as anticipated and to consider using this as a starting point for fixing the query. Once you have validated that the query is working as anticipated, please remove the file with the .fix suffix.

**Comment by jhajduk-microsoft** [comment]
@tksh164 - Thanks so much! That helps with context. I have also submitted the changes as required. Please let me know if further changes are needed. @ejhenry

**Comment by ejhenry** [comment]
> @tksh164 - Thanks so much! That helps with context. I have also submitted the changes as required. Please let me know if further changes are needed. @ejhenry

@jhajduk-microsoft - The query as currently written does not return Standard sku PIPs that are not zone-redundant. We need to catch these as well to identify them for customers as not being aligned with the recommendation.

**Comment by jhajduk-microsoft** [comment]
@ejhenry - I updated and pushed my branch - is it now showing?

**Comment by jhajduk-microsoft** [comment]
@ejhenry - NVM, got it, updating now

**Comment by ejhenry** [comment]
@jhajduk-microsoft - there a three PIP configs we need to return with the query that do not align with the recommendation:

1. Basic PIP
2. Standard PIP - no zone
3. Standard PIP - single zone

The current version of the query is only catching the second config on the list.

**Comment by jhajduk-microsoft** [comment]
Thanks for clearing that up. Here is the screenshot:
image

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Comment by tksh164** [comment]
@jhajduk-microsoft - Could you change the column name `zones` to `param2`? because `zones` is not valid column name.

- [Azure Resource Graph (ARG) Queries - Contribution guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing/#azure-resource-graph-arg-queries)
> 6. ARG query columns name returned should only include the following:

**Comment by jhajduk-microsoft** [comment]
I have updated the param:
image

**Review by rodrigosantosms — COMMENTED** [review]
Approved. Improvements will be made by the team to improve the query later.

**Pull request merged**